packages/@lwc/engine-dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -53,8 +53,10 @@ const hydratedCustomElements = new WeakSet<Element>();
 
 export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLElementConstructor {
     const HtmlPrototype = getComponentHtmlPrototype(Ctor);
+    const { observedAttributes } = HtmlPrototype as any;
+    const { attributeChangedCallback } = HtmlPrototype.prototype as any;
 
-    return class extends HtmlPrototype {
+    return class extends HTMLElement {
```

Comment:
this is definitely wrong. if you do this, then you loose the ability to have a public method. Extending `HtmlPrototype` (which btw is a pretty bad name for that - my fault), allows LWC to define the shape of the element. Now, how those this work? maybe because createVM and hydrateComponent will alter the proto chain of the instance, but it is observable, because you can always inspect the Ctor, and get the public API, which is not the case now. There might be other problems... but ultimate, Core is trying to build the shape, so we can use it. Why did you decide to change this?

Reply:
@caridy This was the only way to get the tests to pass. If I change it back, I get 6 test failures. We can dig into it if you'd like.

Reply:
OK after discussion, we probably need to fix this as well as the constructor issues described here: https://github.com/salesforce/lwc/issues/2877

Also `customElements.get()` and customElements.whenDefined()` should not leak LWC component constructors at all; that's something I missed. (Should return undefined.)

Reply:
Looks like something in the `HTMLBridgeElement` is busted. It throws an error `Illegal constructor` if I replace `HTMLElement` with `HtmlPrototype` above.

![Screen Shot 2022-06-15 at 3 28 52 PM](https://user-images.githubusercontent.com/283842/173942862-059493c1-d32b-462c-b972-567726ef34f2.png)

This is the line that throws from the user perspective:

![Screen Shot 2022-06-15 at 3 32 40 PM](https://user-images.githubusercontent.com/283842/173958997-1524787e-32d9-4706-9c79-63f0fff774ae.png)

I still need to dig in more to figure it out.

Reply:
This turned out to be a pre-existing issue: https://github.com/salesforce/lwc/issues/2970